modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/GlassSystemMenu.java

```diff
@@ -169,32 +208,14 @@ private void insertMenu(final Menu parent, final MenuBase mb, int pos) {
                     }
                 }
             }
-        });
-
-        for (MenuItemBase item : items) {
-            if (item instanceof MenuBase) {
-                // submenu
-                addMenu(glassMenu, (MenuBase)item);
-            } else {
-                // menu item
-                addMenuItem(glassMenu, item);
-            }
-        }
-        glassMenu.setPixels(getPixels(mb));
-
-        setMenuBindings(glassMenu, mb);
-
-        if (parent != null) {
-            parent.insert(glassMenu, pos);
-        } else {
-            glassSystemMenuBar.insert(glassMenu, pos);
-        }
+        };
     }
 
+
     private void setMenuBindings(final Menu glassMenu, final MenuBase mb) {
-        mb.textProperty().addListener(valueModel -> glassMenu.setTitle(parseText(mb)));
-        mb.disableProperty().addListener(valueModel -> glassMenu.setEnabled(!mb.isDisable()));
-        mb.mnemonicParsingProperty().addListener(valueModel -> glassMenu.setTitle(parseText(mb)));
+        mb.textProperty().addListener(new WeakInvalidationListener(valueModel -> glassMenu.setTitle(parseText(mb))));
+        mb.disableProperty().addListener(new WeakInvalidationListener(valueModel -> glassMenu.setEnabled(!mb.isDisable())));
+        mb.mnemonicParsingProperty().addListener(new WeakInvalidationListener(valueModel -> glassMenu.setTitle(parseText(mb))));
```

Comment:
Isn't this the classic mistake being made with weak listeners where you are wrapping a listener, but then are not referring it?  The inner listener (which is created on the fly) is only weakly referenced, and so can go out of scope immediately.